src/java/org/apache/cassandra/db/compression/CompressionDictionaryDetailsTabularData.java

```diff
@@ -0,0 +1,304 @@
+     * @param compositeData data to create data object from
+     * @return deserialized composite data to convenience object
+     * @throws IllegalArgumentException if values in deserialized object are invalid.
+     * @see CompressionDictionaryDataObject#validate()
+     */
+    public static CompressionDictionaryDataObject fromCompositeData(CompositeData compositeData)
+    {
+        String keyspace = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KEYSPACE_NAME);
+        String table = (String) compositeData.get(CompressionDictionaryDetailsTabularData.TABLE_NAME);
+        long dictId = (Long) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_ID_NAME);
+        byte[] dictionaryBytes = (byte[]) compositeData.get(CompressionDictionaryDetailsTabularData.DICT_NAME);
+        String kind = (String) compositeData.get(CompressionDictionaryDetailsTabularData.KIND_NAME);
+        int checksum = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.CHECKSUM_NAME);
+        int size = (Integer) compositeData.get(CompressionDictionaryDetailsTabularData.SIZE_NAME);
+
+        CompressionDictionaryDataObject dataObject = new CompressionDictionaryDataObject();
+        dataObject.keyspace = keyspace;
+        dataObject.table = table;
+        dataObject.dictId = dictId;
+        dataObject.dict = dictionaryBytes;
+        dataObject.kind = kind;
+        dataObject.dictChecksum = checksum;
+        dataObject.dictLength = size;
+
+        dataObject.validate();
+
+        return dataObject;
+    }
+
+    public static class CompressionDictionaryDataObject implements Cloneable
+    {
+        public String keyspace;
+        public String table;
+        public long dictId;
+        public byte[] dict;
+        public String kind;
+        public int dictChecksum;
+        public int dictLength;
+
+        /**
+         * An object of this class is considered to be valid if:
+         *
+         * <ul>
+         *     <li>keyspace and table are not null</li>
+         *     <li>dict id is lower than 0</li>
+         *     <li>dict is not null nor empty</li>
+         *     <li>dict lenght is less than or equal to 1MiB</li>
+         *     <li>kind is not null and there is such {@link org.apache.cassandra.db.compression.CompressionDictionary.Kind}</li>
+         *     <li>dictLenght is bigger than 0</li>
+         *     <li>dictLenght has to be equal to dict's length</li>
+         *     <li>dictChecksum has to be equal to checksum computed as part of this method</li>
+         * </ul>
+         */
+        public void validate()
+        {
+            if (keyspace == null)
+                throw new IllegalArgumentException("keyspace not specified");
+            if (table == null)
+                throw new IllegalArgumentException("table not specified");
+            if (dictId <= 0)
+                throw new IllegalArgumentException("Provided dictionary id is lower than 0, it is '" + dictId + "'.'");
```

Comment:
The condition is `dictId <=0`, but the error message says `lower than 0`. The message is incorrect. Maybe change to "Dictionary id must be positive. The provided is {}"

The ending `"'.'"` seems wrong too.